### ISSUE TITLE
Data management questions

### ISSUE THREAD

**Issue by emlys** [natcap/softwareteam#118]
## Context
At the natcap event at Huddart Park yesterday, there was a discussion about the pros/cons of global conservation priority mapping. Some points I heard were:
- Poor access to data inputs and outputs used in published research e.g. journal article links to a google drive folder but the link is broken; many don't publish the data at all
- Need to emphasize limitations of our models and data products
    - This is more important with global data - we don't want to displace local data that should probably be used instead for various scientific and practical reasons
    - Interest in developing wording for a disclaimer/core principles about using global data/contexts in which it's appropriate (scientific: scale, accuracy; practical: trust, effective communication with local stakeholders
- Need to document what input data was used to create a data product - LULC especially since models are sensitive to LULC changes and some widely used LULC products have large inaccuracies - inappropriate to smaller scale

I brought up our discussion of data APIs (new priority for this quarter) and mentioned versioning and metadata as angles to address these issues in our own global data products, if/when we make them:
- versioning/version control can free us up to make changes
    - a "living" product instead of/in addition to a final, journal published map
    - avoid giving the impression that anything is the one true map
- distribute our documentation/limitations alongside the data
    - within the file metadata (may not be possible), or
    - just include in README, etc and emphasize in UI

This seemed to reflect broader issues folks have experienced with data and there was interest in some sort of software-data-management solution for NatCap, as well as institutionalizing a process for it. 

Gretchen expressed interest in an initial survey of data management options, on the scale of 10 hours working time.

## My interpretation

Research data management is an issue in every field and surely many efforts have already been made to solve it. However it's far from solved and most published research is not providing all their data in a truly open source, reliable way. It would be great if NatCap can be at the forefront of doing this right. This may be a little out of the scope of software, but we're familiar with the issues listed below, so we're well positioned to help other natcappers find/use best practices.

To benefit from the network effect, and avoid reinventing the wheel, we should prioritize using an existing, well-established solution rather than creating our own, as long as that is possible.

Things this solution might do:
- data versioning: a file name should not be the only identifier of a dataset
- version control:
    - access different versions of the data
    - know which version you have
    - make and track changes 
- hosting: the data lives on a remote server and not just on someone's laptop
    - anyone with the link can access (open source science, don't need to contact natcap for access, also works in china)
    - reliable, not dependent on a single person or lab 
    - provides permalinks
- distribution/uptake: 
    - links to data can be included in published articles
    - broadly supported and documented
    - easy to use without having to learn something new
    - the interface does not scare people away

## Action Items

- Conduct a survey of existing options to address some/all of the points above
- Talk to lead scientists before publishing global data via APIs. We may need to use discretion about which models' outputs we publish and how we present them
-  Consider a simple scale disclaimer for web viewer. For example, in this [USDA soil data viewer](https://websoilsurvey.sc.egov.usda.gov/App/WebSoilSurvey.aspx), if you use the "identify" option when zoomed in too much, it shows a "Warning: Soil Map may not be valid at this scale."
- Consider building a disclaimer about the other issues discussed into invest/web viewer/other products and emphasize limitations more in users guide. Science team would provide wording for this.

## Primary Contacts

Lisa and Kelley (and probably a couple others I'm forgetting) specifically expressed interest in following up about this.

---

## Software Team Section

- [ ] Roughly how long is the requested work estimated to take given the above scope?
- [ ] Given current team priorities, when might the team be able to take this on?

**Comment by phargogh** [comment]
Yeah, data management has been a longstanding question at NatCap.  Certainly this relates to our Data APIs strategic goal, and also to the Data Platform as a component of the NatCap Platform.

From talking with @lmandle about this, she thought that the conversation was mostly focused around internal documentation and preservation of outputs ... a good set of documentation and guidance (perhaps with some recommended software tools) might be a sufficient solution.  In addition to the names above, she mentioned Jesse since he had previously been managing some largish datasets for NCES (sp?).

If the internal solution is pursued, it might end up answering some of the data platform questions and a data platform might be a natural iteration on it.

Sounds like Lisa will be starting a slack message/group on this subject to continue the discussion among those who were interested in being involved from the Huddart conversation. (and @emlys she has the large sheet of paper you were wondering about!)

So, I think the ~10 hours proposed is a really good idea, so we know what's out there as far as tooling goes!  And on that subject, we should also consider looking at MyGeoHub (which is a software platform that's come up through the GLASSNET meetings and working towards #107 )

Also, it sounds like there may be some funding in this (there's some internal Stanford funding that may be accessible for this sort of thing), so let's not discard a promising solution for hosting costs.  If MyGeoHub proves interesting, there may be some funding in that as well.
